src/libvirtApi/domain.js

```diff
@@ -547,7 +548,10 @@ export function domainGet({
 
                 snapshotGetAll({ connectionName, domainPath: objPath });
             })
-            .catch(ex => console.warn("GET_VM action failed for path", objPath, ex.toString()));
+            .catch(ex => {
+                console.warn("GET_VM action failed for path", objPath, ex.toString());
+                store.dispatch(undefineVm({ connectionName, id: objPath }));
```

Comment:
If this is too much/wide, we could alternatively let domainGet() return the promise, and leave the error handling to all callers. @skobyda, @KKoukiou , can you think of a situation where this function would fail in a condition other than the VM actually going away?

Reply:
I filed #518 with that alternative implementation. I can honestly not judge which one would be conceptually better, as I am not sure if a GET_VM is expected under normal circumstances other than during domain deletion.

Reply:
I looked at https://gitlab.com/libvirt/libvirt/-/blob/master/src/conf/domain_conf.c#L28124 which is basically where the magic behind `GetXMLDesc` API happens, and although there are many cases defined when error status is returned, deep down most of them are just result of errors which may happen during string manipulation, when XML string is formatted.
And no, I did not seen a situation when domainGet failed apart from just being non-existent (like in the bug which this fix).

> as I am not sure if a GET_VM is expected under normal circumstances other than during domain deletion. 

What do you mean? If GET_VM function is called in under places apart from deletion, or if just error message is expected to happen in other situations apart from deletion?

Reply:
What gets me a bit more worried is a case when `snapshotGetAll` fails, and as a result of this failure, the VM will be undefined from redux store.

But since the GET_VM in our bug is coming only `GetXMLDesc`, maybe we can limit `store.dispatch(undefineVm({ connectionName, id: objPath }));` only to failures caused by `GetXMLDesc`? And I think it would be safe to just undefine VM from store if `GetXMLDesc` fails. Since if it fails, that means we don't really have any (or updated) data about the VM, and there is nothing to show about the VM in the UI, so practically there is no reason to have it in our redux store.

Reply:
> What do you mean? If GET_VM function is called in under places apart from deletion, or if just error message is expected to happen in other situations apart from deletion?

I mean the latter -- if we e.g. expect GET_VM/GetXMLDesc to fail on creation, in particular in conjuction with the placeholder domains that we are (or were?) creating in some cases. At least the tests don't run into such a situation.

> What gets me a bit more worried is a case when snapshotGetAll fails, and as a result of this failure, the VM will be undefined from redux store.

That's a good point.

> maybe we can limit `store.dispatch(undefineVm({ connectionName, id: objPath }));` only to failures caused by GetXMLDesc?

That's a good idea! I'll change the PR accordingly.

Reply:
I checked the code, and snapshotGetAll() can't really fail. I get this *all the time* on RHEL 8.6, as it's unsupported:

> LIST_DOMAIN_SNAPSHOTS action failed for domain /org/libvirt/QEMU/domain/_9728f5f7_fe69_412a_8c99_51fdc8bb34a8: {"problem":null,"name":"org.freedesktop.DBus.Error.UnknownMethod","message":"No such method 'ListDomainSnapshots'"} index.js:2:1422511

and snapshotGetAll() also just has a catch-all and ignores all errors. But that is bad design -- it should check the error and *only* ignore `UnknownMethod`, not all the others.

But it's still conceptually better to catch GetXML failures only, of course.

Reply:
Updated to your suggestion.

Reply:
This was wrong after all -- it only closed the race condition for the first GetXML call, but the VM could just as well go away on the second one or the GetState(). I'll send another PR.